### DISCUSSION TITLE
ClickHouse's equivalent of the Redshift's CHECKSUM function

### DISCUSSION THREAD

**Discussion by ghost** [ClickHouse/ClickHouse#54605] [Q&A] [answered]
Dear ClickHouse team,

I'm migrating a huge Redshift codebase to ClickHouse and I'm facing the problem of finding the ClickHouse's equivalent of the Redshift's CHECKSUM function https://docs.aws.amazon.com/redshift/latest/dg/r_CHECKSUM.html

Do you know which among the many hash functions of ClickHouse is the equivalent one, if any.

A small test case that I would have to reproduce on ClickHouse would be:

function(ea8a1a99f6c94c275a58dcd78f418c1f) = 11984627
function(11aedd0e432747c2bcd97b82808d24a0) = 1146539

Cheers,

Hugo

**Comment by den-crane** [comment, marked as answer]
It seems this function uses different implementations for different datatypes:

```
select checksum(commission)
from sales
order by salesid
limit 10;

checksum
----------
10920
1140
5250
2625
2310
5910
11820
2955
8865
975
```

These numbers do not look like Int64.
I guess there is no anything even close in Clickhouse.
